hypothesis-python/tests/cover/test_filter_rewriting.py

```diff
@@ -153,17 +156,29 @@ def test_filter_rewriting_floats(data, strategy, predicate, min_value, max_value
 @pytest.mark.parametrize(
     "pred",
     [
+        math.isinf,
+        math.isnan,
         partial(operator.lt, 6),
         partial(operator.eq, Fraction(10, 3)),
-        partial(operator.eq, "can't compare to strings"),
         partial(operator.ge, 0),
         partial(operator.lt, math.inf),
         partial(operator.gt, -math.inf),
     ],
 )
 @pytest.mark.parametrize("s", [st.integers(1, 5), st.floats(1, 5)])
-@fails_with(Unsatisfiable)
 def test_rewrite_unsatisfiable_filter(s, pred):
+    rewritten_strat = s.filter(pred).wrapped_strategy
+    assert isinstance(rewritten_strat, Nothing)
```

Comment:
```suggestion
    assert s.filter(pred).is_empty
```
Just a matter of idiom here.